### ISSUE TITLE
Inconsistencies when a comic fails / ways of contact

### ISSUE THREAD

**Issue by Olf0** [tardypad/sailfishos-daily-comics#38]
Unfortunately "Wizard of ID" ceased to work 2018-04-05.
Its website is working (http://www.creators.com/features/wizard-of-id) and displaying the strips from 2018-04-05 and 2018-04-06, which Daily Comics fails to display.  The URL used for "Wizard of ID" by Daily Comics is also seems to be working (http://www.gocomics.com/wizardofid), but apparently something was changed there.

Daily Comics' behaviour is interesting though:

1. When starting Daily Comics, it detects a new "Wizard of ID" strip and displays "⬇ New" for it.
2. When entering "Wizard of ID" via "Read all new comics", a fully black screen (except for the back button in the upper left corner) is displayed: ![screenshot_20180406_003](https://user-images.githubusercontent.com/16547876/38445382-9e952238-39f3-11e8-8d13-f2759a328fad.png)
3. When entering "Wizard of ID" via its icon, an error message is displayed: ![screenshot_20180406_002](https://user-images.githubusercontent.com/16547876/38430814-a15c37ee-39c2-11e8-8162-2b7ac7c84491.png)
4. After step 3, Daily Comics also displays "❗ Error" on the "Wizard of ID" icon, but returns to displaying "⬇ New" after a while, again (apparently the subscribed comics are rescanned).

Before 2018-04-05 I occasionally experienced "❗ Error" on some subscribed comics when trying to fetch (i.e. download) them, but never when trying to view them!

Expected results:

* Daily Comics behaves identically in steps 2 and 3, displaying an error message (as in step 3).
* "Wizard of ID" either shows "❗ Error" when fetched or displays the actual comic strip (the latter is preferable, IMHO ;) ).

Side note: As Jolla's Mail was not configured and used before on this device (using K9 instead), trying to report via email results in ![screenshot_20180406_001](https://user-images.githubusercontent.com/16547876/38430845-bfe65b7c-39c2-11e8-9ea0-cfc46ef802a2.png)
But I have no idea how to handle this better ([Mimer](https://openrepos.net/content/llelectronics/mimer) unfortunately does not allow to set the default email application).  Maybe an additional way to copy the generated error report including the email address and subject line to the clipboard (as an alternative to calling the email app with these parameters) is feasible.

Tested with Daily Comics 1.7.5-1 under SailfishOS 2.1.4.14 on a Jolla 1 phone.

**Comment by sthocs** [comment]
Hi, indeed GoComics has changed its website.
I've fixed it and submitted a pull request here: #39 
Until it's merged and put on OpenRepos, I've built a version with the fix that you can download here: http://sthocs.free.fr/harbour-dailycomics-1.7.5-2.armv7hl.rpm

**Comment by Olf0** [comment]
@sthocs, thanks a lot for fixing the original issue: GoComics has changed their site layout.
Side note: Kudos for compiling Daily Comics RPM v1.7.5-2, but unfortunately https://sthocs.free.fr/ (via HTTP**S**) permanently runs into connection timeouts for me (in Firefox and `curl`), but http://sthocs.free.fr/harbour-dailycomics-1.7.5-2.armv7hl.rpm works.

As your PR [#39](https://github.com/tardypad/sailfishos-daily-comics/pull/39) addresses solely the original issue, IMO this issue report (#38) should be kept open after merging your PR, until the other bugs are investigated, which have been uncovered by GoComics change:

2. Daily Comics should behave identically in steps 2 and 3, displaying an error message (as in step 3).
3. A comic strip should either show "❗ Error" when failing to fetch its image file (e.g. by testing that an image file type has been successfully received), or (when successful) display the actual comic strip.  But not state  "⬇ New" (which is untrue, as no image file has been successfully fetched) and just display a fully black frame.
Additionally Go Comics 1.7.5-1 behaves as if there is a permanently unread comic strip in this situation, as viewing the black frame does not seem to toggle the "read" flag (which may be resolved in one go by enhancing the error handling).
4. Daily Comics should make the email address for bug reports also available by other means (e.g. in an "About"-page, which currently does not exist) than the aforementioned "contact me, if the problem persists"-button and the corresponding "Let me know ..."-button in its settings, as these fail, when Jolla's email client app is not configured (e.g. when using K9-Mail instead).

**Comment by tardypad** [comment]
Wizard of ID fix is now merged from the PR mentioned before.

I'll try to tackle the other issues for the next release if possible

As for 4, there is an about page at the bottom of the comics list, in a PushUp menu, with all the ways to contact me

**Comment by Olf0** [comment]
@tardypad, thank you.

As for 4, I cannot find it, just the mentioned *"Let me know ..."-button in its settings*, which (also) depends on the Jolla Mail client to be configured: ![screenshot_20180512_001](https://user-images.githubusercontent.com/16547876/39959306-ee3c2380-560f-11e8-9af0-30279c576dc8.png)

Side note: A link to Daily Comics Github issue tracker (https://github.com/tardypad/sailfishos-daily-comics/issues/) IMO would be nice and cause your inbox to be loaded less.

**Comment by tardypad** [comment]
@Olf0 The About page is at the bottom of the main page, not of the settings page as you show

**Comment by Olf0** [comment]
WRT sub-issue 4:
@tardypad, thanks, I missed that bottom push-up menu at Daily Comics main page.
Testing it reveals a couple of similar issues, there:
* **Twitter icon**
  Pressing it opens *https://twitter.tardypad.me/* in the default browser, which immediately runs into a certificate error: *twitter.tardypad.me uses an invalid security certificate. The certificate is only valid for the following names: busy.tardypad.me, pkgs.tardypad.me, tardypad.me, www.tardypad.me Error code: SSL_ERROR_BAD_CERT_DOMAIN*
  Insisting to visit this web-page (by ignoring the certificate error) results in: *https://twitter.tardypad.me/ is requesting your user name and password. The site says: "Restricted area"*
  Ultimately and logically a ***401 Authorisation required*** is displayed.
* **Github icon**
  Pressing it opens *https://github.tardypad.me/* in the default browser, which immediately runs into a certificate error: *github.tardypad.me uses an invalid security certificate. [...]* (see bullet point above).
  Insisting to visit this web-page (by ignoring the certificate error) results in: *https://github.tardypad.me/ is requesting your user name and password. The site says: "Restricted area"*
  Ultimately and logically a ***401 Authorisation required*** is displayed.
* **Email icon**
  Sub-issue 4 applies, if not using the Jolla email-client.

Hence no way of contact works for me.

IMO and due to @maledictus seemingly having taken over the maintenance of Daily Comics, this should be generalised by simply providing a few web-links to relevant locations.
Thus filing this as sub-issue 5.

Sure, I concur that this bottom menu at Daily Comics' main page is the right place to display, e.g. (as possible solutions of sub-issue 5):
* OpenRepos page (installable RPM): https://openrepos.net/content/tardypad/daily-comics
* Project page (source code): https://github.com/tardypad/sailfishos-daily-comics
  (This web-link is already there, but embedded in a small block of text.)
* Issue tracker: https://github.com/tardypad/sailfishos-daily-comics/issues

Also sub-issue 4 can be resolved the same way: By a link to Daily Comics' issue tracker (while its text and location at the bottom of the settings page is perfect and should be retained).

**Comment by Maledictus** [comment]
I have got write permissions to this repository and I can continue development of this project but I haven't access to openrepos account and @tardypad said that he will publish releases himself

**Comment by Olf0** [comment]
@maledictus, I already gathered this [from TJC](https://together.jolla.com/question/195152/adopting-abandoned-apps/) and saw you committing here: Thank you very much!
It is nice to have the chance to express that properly (in addition to a *+1* at TJC), but OTOH this is not really related to this issue.

But while at this topic: You may still make releases and upload packaged RPMs at https://github.com/tardypad/sailfishos-daily-comics/releases (i.e., without @tardypad's help).

**Comment by tardypad** [comment]
I'm closing this issue which had become a list of different unrelated problems and requests.
AFAICS the only problem left here is related to the handling of errors for which I just created a distinct issue (linking it to here if as a reference)

**Comment by Olf0** [comment]
@tardypad, thanks.
I absolutely agree, that this issue has been overloaded (mostly by me) with a bunch of issues, which are related in terms of user's workflow, but are technically unrelated.
